fabric-enchanting-api-v1/src/main/java/net/fabricmc/fabric/mixin/enchanting/MixinEnchantingTableContainer.java

```diff
@@ -34,34 +31,22 @@
 
 @Mixin(EnchantingTableContainer.class)
 public abstract class MixinEnchantingTableContainer {
-	@Unique
-	private World world;
-	@Unique
-	private BlockPos blockPos;
-
-	@SuppressWarnings("UnresolvedMixinReference")
-	@Inject(method = "method_17411(Lnet/minecraft/item/ItemStack;Lnet/minecraft/world/World;Lnet/minecraft/util/math/BlockPos;)V", at = @At("HEAD"))
-	private void onEnchantmentCalculation(ItemStack itemStack, World world, BlockPos blockPos, CallbackInfo callbackInfo) {
-		this.world = world;
-		this.blockPos = blockPos;
-	}
-
 	@SuppressWarnings("UnresolvedMixinReference")
 	@ModifyVariable(method = "method_17411(Lnet/minecraft/item/ItemStack;Lnet/minecraft/world/World;Lnet/minecraft/util/math/BlockPos;)V", at = @At(value = "INVOKE", target = "Ljava/util/Random;setSeed(J)V", shift = At.Shift.BEFORE), ordinal = 0)
-	private int changeEnchantingPower(int power) {
+	private int changeEnchantingPower(int power, World world, BlockPos blockPos) {
```

Comment:
You need the item stack in the signature or else mixin will fail to apply

Reply:
sorry